tests/MassTransit.KafkaIntegration.Tests/Avro_Specs.cs

```diff
@@ -50,21 +40,16 @@ static IDeserializer<T> GetDeserializer<T>(IServiceProvider provider)
                     {
                         rider.AddConsumer<TestKafkaMessageConsumer<KafkaMessage>>();
 
-                        rider.AddProducer<string, KafkaMessage>(Topic, context => context.MessageId.ToString(), (context, cfg) =>
-                        {
-                            cfg.SetKeySerializer(GetSerializer<string>(context));
-                            cfg.SetValueSerializer(GetSerializer<KafkaMessage>(context));
-                        });
+                        rider.AddProducer<string, KafkaMessage>(Topic, context => context.MessageId.ToString());
 
                         rider.UsingKafka((context, k) =>
                         {
+                            k.SetSerializationFactory(context.GetRequiredService<IKafkaSerializerFactory>());
```

Comment:
This "gets it done" and leaves it up to the developer I guess, but really exposes the underlying container.

Reply:
exactly